pdns/qtype.cc

```diff
@@ -30,23 +30,79 @@
 
 static_assert(sizeof(QType) == 2, "QType is not 2 bytes in size, something is wrong!");
 
-vector<QType::namenum> QType::names;
-// XXX FIXME we need to do something with initializer order here!
-QType::init QType::initializer; 
-
-QType::QType()
+const vector<QType::namenum> QType::names = {
```

Comment:
I wonder if we could wrap this into `boilerplate_conv` via DNSRecords. Or make `boilerplate_conv` take a QType, so we never have the problem of types not being in sync with supported types?

Reply:
I changed the boilerplate so that it uses QType::RNAME, enforcing it must exists. That's a start, I'd say.